### PR TITLE
async image loading from data, files and resources for ios

### PR DESCRIPTION
**Pull request by enchev**

### CODE REVIEW

File: image-source/image-source-common.ts [hunk 1]
```diff
@@ -11,45 +11,83 @@ function ensureHttp() {
 // This is used for definition purposes only, it does not generate JavaScript for it.
 import definition = require("image-source");
 
-export function fromResource(name: string): definition.ImageSource {
+export function fromResource(name: string, async?: boolean): definition.ImageSource | Promise<definition.ImageSource> {
     var image = new definition.ImageSource();
-    return image.loadFromResource(name) ? image : null;
+    var result : any = image.loadFromResource(name, async);
+    if (result instanceof Promise) {
+        return new Promise<definition.ImageSource>((resolve, reject) => {
```
Comment (PanayotCankov) [inline image-source/image-source-common.ts]: Why don't we return "result" here?

File: image-source/image-source-common.ts [hunk 2]
```diff
@@ -11,45 +11,83 @@ function ensureHttp() {
 // This is used for definition purposes only, it does not generate JavaScript for it.
 import definition = require("image-source");
 
-export function fromResource(name: string): definition.ImageSource {
+export function fromResource(name: string, async?: boolean): definition.ImageSource | Promise<definition.ImageSource> {
```
Comment (PanayotCankov) [inline image-source/image-source-common.ts]: Won't be a `fromResourceAsync` better?

**Comment by ErjanGavalji** [comment]
:+1:

**Comment by enchev** [comment]
Unfortunately due to thread-safe problems I'm closing this:
http://stackoverflow.com/questions/12766501/is-creating-uiimage-thread-safe

**Pull request closed without merging**